liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -0,0 +1,882 @@
+            
+            this.existingDependencies = project.getDependencies();
+            File pom = project.getFile();
+            this.existingPom = readFile(pom);
+            ServerFeature servUtil = getServerFeatureUtil();
+            this.existingFeatures = servUtil.getServerFeatures(serverDirectory);
+        }
+
+        @Override
+        public void debug(String msg) {
+            log.debug(msg);
+        }
+
+        @Override
+        public void debug(String msg, Throwable e) {
+            log.debug(msg, e);
+        }
+
+        @Override
+        public void debug(Throwable e) {
+            log.debug(e);
+        }
+
+        @Override
+        public void warn(String msg) {
+            log.warn(msg);
+        }
+
+        @Override
+        public void info(String msg) {
+            log.info(msg);
+        }
+
+        @Override
+        public void error(String msg) {
+            log.error(msg);
+        }
+
+        @Override
+        public boolean isDebugEnabled() {
+            return log.isDebugEnabled();
+        }
+
+        @Override
+        public void stopServer() {
+            try {
+                ServerTask serverTask = initializeJava();
+                serverTask.setOperation("stop");
+                serverTask.execute();
+            } catch (Exception e) {
+                // ignore
+                log.debug("Error stopping server", e);
+            }
+        }
+
+        @Override
+        public void startServer() {
+
+            try {
+                // Setup server task
+                if (serverTask == null) {
+                    serverTask = initializeJava();
+                }
+
+                String logsDirectory = serverTask.getOutputDir() + "/" + serverTask.getServerName() + "/logs";
+                messagesLogFile = new File(logsDirectory + "/messages.log");
+
+                copyConfigFiles();
+                serverTask.setClean(clean);
+                serverTask.setOperation("debug");
+
+
+                // Set server start timeout
+                if (serverStartTimeout < 0) {
+                    serverStartTimeout = 30;
+                }
+                serverTask.setTimeout(Long.toString(serverStartTimeout * 1000));
+
+
+                // Watch logs directory if it already exists
+                WatchService watchService = FileSystems.getDefault().newWatchService();
+                boolean logsExist = new File(logsDirectory).isDirectory();
+
+                if (logsExist) {
+                    // If the logs directory already exists, then
+                    // setup a watch service to monitor the directory.
+                    Paths
+                        .get(logsDirectory)
+                        .register(watchService, StandardWatchEventKinds.ENTRY_CREATE, StandardWatchEventKinds.ENTRY_MODIFY);
+                }
+
+
+                // Start server
+                Thread serverThread = new Thread(new Runnable() {
```

Comment:
Why does the server task need to be executed from a new thread?

Reply:
We are running the server in debug mode which runs in the foreground.  We need to keep that in a separate thread so that while debug mode is running in that thread, we can continue to do things such as check for file changes and recompiling the code.